src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -2580,6 +2580,42 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::FromOtDnsRespons
     return CHIP_NO_ERROR;
 }
 
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::DispatchBrowseInvokedAddressResolve(intptr_t context)
+{
+    DnsResult * dnsResult = reinterpret_cast<DnsResult *>(context);
+
+    ThreadStackMgrImpl().LockThreadStack();
+
+    const otDnsQueryConfig * defaultConfig = otDnsClientGetDefaultConfig(ThreadStackMgrImpl().OTInstance());
+
+    char fullHostName[chip::Dnssd::kHostNameMaxLength + 1 + SrpClient::kDefaultDomainNameSize + 1];
+    snprintf(fullHostName, sizeof(fullHostName), "%s.%s", dnsResult->mMdnsService.mHostName, SrpClient::kDefaultDomainName);
+
+    otDnsClientResolveAddress(ThreadStackMgrImpl().OTInstance(), fullHostName, OnDnsBrowseInvokedAddressResolveResult,
+                              reinterpret_cast<void *>(dnsResult), defaultConfig);
+
+    ThreadStackMgrImpl().UnlockThreadStack();
+}
+
+template <class ImplClass>
+void GenericThreadStackManagerImpl_OpenThread<ImplClass>::DispatchAddressResolve(intptr_t context)
+{
+    DnsResult * dnsResult = reinterpret_cast<DnsResult *>(context);
```

Comment:
As only the callback function is different, would it make sense to have one common method as you did with `OnDnsAddressResolveResult`?

Reply:
I think I could create a wrapper implementing most of the functionality and taking callback name as an input, but still I guess we would need two different dispatch methods to pass the appropriate callbacks.